core/metadata_vec.h

```diff
@@ -0,0 +1,631 @@
+/***************************************************************************
+ *
+ * Authors:     Jan Horacek (xhorace4@fi.muni.cz)
+ *
+ * Institute of Computer Science MUNI
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+
+#ifndef VEC_METADATA_H
+#define VEC_METADATA_H
+
+#include <memory>
+#include <exception>
+#include "memory.h"
+#include "metadata_base.h"
+#include "metadata_base_it.h"
+#include "metadata_row_vec.h"
+
+class NotImplemented : public std::logic_error
+{
+public:
+    NotImplemented() : std::logic_error("Function not yet implemented") { };
+};
+
+
+using MetaDataVecRow = std::vector<MDObject>;
```

Comment:
don't put `using` in global namespace in headers

Reply:
Why? This is so specific name that collision will never happen. This is like arguing *do not create class MySuperClass* in global namespace.

This is not `using namespace std`, this is just a modern `typedef`.